tikv-client/src/main/java/com/pingcap/tikv/allocator/RowIDAllocator.java

```diff
@@ -125,20 +130,30 @@ public long getEnd() {
     return end;
   }
 
-  // set key value pair to tikv via two phase committer protocol.
-  private void set(ByteString key, byte[] value) {
+  // set key value pairs to tikv via two phase committer protocol.
+  private void set(List<BytePairWrapper> pairs) {
+    if (pairs == null || pairs.size() == 0) {
+      return;
+    }
     TiSession session = TiSession.getInstance(conf);
     TwoPhaseCommitter twoPhaseCommitter =
```

Comment:
you can write it like this
      Iterator<BytePairWrapper> iterator = pairs.iterator();
      BytePairWrapper primaryPair = iterator.next();
And use iterator directly in the next to rewrite secondary keys.